src/app/reporting/EventLoggingTypes.h

```diff
@@ -0,0 +1,273 @@
+ * event of high priority is added to a full buffer, events are
+ * dropped in order of priority (and age) to accommodate it. As such,
+ * priority levels only have relative value. If a system is
+ * using only one priority level, events are dropped only in order
+ * of age, like a ring buffer.
+ */
+
+typedef enum PriorityLevel
+{
+
+    kPriorityLevel_First = 1,
+    /**
+     * Critical priority denotes events whose loss would
+     * directly impact customer-facing features. Applications may use
+     * loss of Production Critical events to indicate system failure.
+     * On constrained devices, entries logged with Critical
+     * priority must be accounted for in the power and memory budget,
+     * as it is expected that they are always logged and offloaded
+     * from the device.
+     */
+    Critical = kPriorityLevel_First,
+    /**
+     * Info priority denotes log entries that provide extra insight
+     * and diagnostics into the running system. Info logging level may
+     * be used over an extended period of time in a production system,
+     * or may be used as the default log level in a field trial. On
+     * the constrained devices, the entries logged with Info level must
+     * be accounted for in the bandwidth and memory budget, but not in
+     * the power budget.
+     */
+    Info = 2,
+
+    /**
+     *  Debug priority denotes log entries of interest to the
+     *  developers of the system and is used primarily in the
+     *  development phase. Debug priority logs are
+     *  not accounted for in the bandwidth or power budgets of the
+     *  constrained devices; as a result, they must be used only over
+     *  a limited time span in production systems.
+     */
+    Debug                  = 3,
+    kPriorityLevel_Last    = Debug,
+    kPriorityLevel_Invalid = 4,
+
+} PriorityLevel;
+
+/**
+ * @brief
+ *   The structure that defines a schema for event metadata.
+ */
+struct EventSchema
+{
+    chip::ClusterId mClusterId; ///< ID of cluster
+    uint32_t mStructureType;    ///< Type of structure
+    PriorityLevel mPriority;    ///< Priority
+};
+
+/**
+ * @typedef timestamp_t
+ * Type used to describe the timestamp in milliseconds.
+ */
+typedef uint32_t timestamp_t;
+
+/**
+ * @typedef duration_t
+ * Type used to describe the duration, in milliseconds.
+ */
+typedef uint32_t duration_t;
+
+/**
+ * @typedef utc_timestamp_t
+ * Type used to describe the UTC timestamp in milliseconds.
+ */
+typedef uint64_t utc_timestamp_t;
+
+enum
+{
+    kTag_Region  = 1, ///< An 32-bit unsigned indicating the log region, i.e. the module to which the log message pertains.
+    kTag_Message = 2  ///< A string containing the actual debug message.
+};
+
+// Structures used to describe in detail additional options for event encoding
+
+/**
+ * @brief
+ *   The structure that provides a full resolution of the cluster.
+ */
+struct DetailedRootSection
+{
+    /**
+     * Default constructor
+     */
+    DetailedRootSection(void){};
```

Comment:
delete this constructor and comment - it does not seem to help readability

Reply:
done, thanks